src/coreclr/jit/optcse.cpp

```diff
@@ -3066,8 +3066,11 @@ class CSE_Heuristic
 
                                 assert(vnStore->IsVNCompareCheckedBoundArith(oldCmpVN));
                                 vnStore->GetCompareCheckedBoundArithInfo(oldCmpVN, &info);
```

Comment:
What about the other uses of this function? Do they need to be changed too?

Reply:
@jakobbotsch I've inspected all uses and didn't find places where we might make wrong decisions due to order of operands (the only use of `arrOp` is this CSE where we compose a new VN and in rangecheck)